File: Data Structures & Algorithms/validate-parentheses/submission-12.java
class Solution {
    public boolean isValid(String s) {
        
        if(s.length() % 2 != 0) return false;

        Stack<Character> stack = new Stack<>();

        for(int i = 0; i < s.length(); i++) {

            if(s.charAt(i) =='(') {
                stack.push(')');
            }

            if(s.charAt(i) =='{') {
                stack.push('}');
            }

            if(s.charAt(i) =='[') {
                stack.push(']');
            }

            if(s.charAt(i) == ')' || s.charAt(i) == '}' || s.charAt(i) == ']') {
                if(stack.isEmpty() || stack.pop() != s.charAt(i)) return false;
            }

        }

        return stack.isEmpty();
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-12.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:

        stack = []

        for token in tokens:

            # Edge case where stack is empty when computing
            if not stack and token in ["+", "-", "*", "/"]:
                return

            match token:
                case '+':
                    second, first = stack.pop(), stack.pop()
                    stack.append(first + second)
                case '-':
                    second, first = stack.pop(), stack.pop()
                    stack.append(first - second)

                case '*':
                    second, first = stack.pop(), stack.pop()
                    stack.append(first * second)

                case '/':
                    second, first = stack.pop(), stack.pop()
                    stack.append(int(first / second))
                case _:
                    stack.append(int(token))

        return stack[-1]